sympy/physics/mechanics/kane.py

```diff
@@ -524,6 +528,9 @@ def kanes_equations(self, bodies, loads=None):
             Must be either a non-empty iterable of tuples or None which corresponds
             to a system with no constraints.
         """
+        if sub_kin_diffs:
+            self.sub_kin_diffs = True
```

Comment:
Can't this just be: `self.sub_kin_diffs = sub_kin_diffs`? I also recommend making the attribute private (`self._sub_kin_diffs`).